maint/gen_binding_f08.py

```diff
@@ -0,0 +1,122 @@
+##
```

Comment:
> [Critical] `status_c` is undefined in `MPI_Get_count_f08()`, `MPI_Get_elements_x_f08()`, `MPI_Test_cancelled_f08()` and possibly some other functions. For example, the following does not initialize `status_c`.

Missing `status_c = status` in the else banch, right?

Reply:
Missing status_c = status in the else banch, right?

Reply:
> [Not critical] Can we unify the capitalization? They are mixed. For example:
> 
> ```
> TYPE/INTEGER/LOGICAL->type/integer/logical
> ```
> 
> I know Fortran accepts both, but if the code is autogenerated, the same format is favorable.

Since the MPI standard uses capital keywords, I converted most Fortran keywords to Capitals now.

Reply:
> [Minor] Some functions remove the `c_int == kind(0)` branch for some reasons. For example, `MPI_Alltoallw_f08ts()`, `MPI_Ialltoallw_f08ts()`, `MPI_Comm_create_errhandler_f08()`, `MPI_Add_error_class_f08()` ... and many. I believe the resulting code is correct (just need extra copy when `c_int = kind(0)`), but I do not understand why it needs to be removed. I would like to keep it if possible.

Fixed.

Reply:
> [Minor] Can we unify the array assignment format?

Unfortunately, it is more complicated and the patterns have subtle differences. When the array dimension is known in the type declaration, Fortran can directly do array assignment. In the case of assumed length (that the dimension is unknown at compile time but known with another parameter), the dimension need be provided. In principle, either the left side dimension or the right side dimension is sufficient, but it simpler for the script to generate for both anyway. Then in the case of "automatic reallocation" of allocatable array, the dimension need be given on the right side so the compiler would know the left dimension to allocate the new array.

In the name of making programming simpler (by providing array syntax) I think it made the code rather complex :(

Reply:
Got it. Then no need to fix this.

Reply:
> Minor] I think it should be `index`, not `indx` (at least the spec spells out `index`).

This is the side effect of the binding interface (to work with C interface). In C, index is a standard library function so (some) compilers will issue `-Wshadow` warning if local variables are named index, thus, the binding scripts always uses `indx` for `index`. Fortran is okay, but rather than add another conversion back, let's just keep it simple.

Reply:
I didn't know that. Thanks!